### ISSUE TITLE
Spacemacs and fish shell do not get along

### ISSUE THREAD

**Issue by zifeo** [syl20bnr/spacemacs#5593]
#### Description
Weird chars appearing at random when using fish shell. Whenever those chars appear, deleting them results will get rid the command line (on enters) but they will still be present in the prompt when typing. No issue when using bash.

#### Reproduction guide
- Start Emacs
- `<SPC> '`
- start typing around

*Observed behaviour:*
![img_2016-03-25 17 11 45](https://cloud.githubusercontent.com/assets/9053709/14048153/ae8f9848-f2ac-11e5-8945-02b0d6f0b30b.jpg)

*Expected behaviour:*
No weird chars.

#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.14
- Spacemacs branch: master (rev. 3aff734)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(auto-completion emacs-lisp git markdown
                 (scala :variables scala-auto-insert-asterisk-in-comments t)
                 (osx :variables osx-use-option-as-meta nil)
                 org
                 (shell :variables shell-default-height 30 shell-default-position 'bottom)
                 (spell-checking :variables spell-checking-enable-auto-dictionary t)
                 syntax-checking spotify)
```

**Comment by d12frosted** [comment]
The same here. But don't know how to solve it.

**Comment by JAremko** [comment]
Try setting `TERM=xterm-old` it should make fish less  :rainbow:

**Comment by zifeo** [comment]
Doesn't change anything :disappointed:

**Comment by zifeo** [comment]
@StreakyCobra is not this considered as a bug :/ ?

**Comment by StreakyCobra** [comment]
@zifeo Don't worry to much about labels, it's mainly for us, and it can change as information arrive :-) As it is now I'm not sure it's a Spacemacs bug so I've put forum for now. It may be a misconfiguration of your terminal, like TERM or SHELL variable not correctly set, or some weird prompt config. Let see how it evolves (people encoutering it, or helping you to identify the cause, etc…)

In this case it's probably an upstream problem anyway, so :smiley:

**Comment by JAremko** [comment]
How about this ? fish-shell/fish-shell#107

**Comment by JAremko** [comment]
I think [ansi-term](https://www.emacswiki.org/emacs/AnsiTerm)  look ok. Except return codes.

**Comment by zifeo** [comment]
Thanks @StreakyCobra for the info and @JAremko for the pointer! 
```fish
tic -o ~/.terminfo $TERMINFO/e/eterm-color.ti
```
From inside spacemacs seems to have worked, weird behavior from ansi-term though.

**Comment by JAremko** [comment]
@zifeo and @d12frosted  It might be a good idea to add a note with the solution to the [shell layer README](https://github.com/syl20bnr/spacemacs/tree/develop/layers/%2Btools/shell)   Not sure if it's the best place. Might be @StreakyCobra  has an idea.

**Comment by StreakyCobra** [comment]
FAQ it :-)

**Comment by everson** [comment]
I am also having issues where Spacemacs and Fish does not play well together, but with a different behavior. In my case, as I type, sometimes the commands are overlapped with previous output and some numbers are displayed in the next lines as shown on http://take.ms/1Qud8

I have [oh-my-fish](url) configured. I removed oh-my-fish and the problem went away. It would still be nice to be able to keep oh-my-fish, though. I am leaving this in hopes that someone knows how to fix it and also to those having the same issue to know that disabling oh-my-fish solves the issue.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
